libraries/chain/wasm_alloc_pool.cpp

```diff
@@ -8,31 +8,24 @@ wasm_alloc_pool::wasm_alloc_pool()
    , max_call_depth(1)
 {
    // create 1 wasm allocator for the main thread
-   stack.push(new vm::wasm_allocator);
-}
-
-wasm_alloc_pool::~wasm_alloc_pool() {
-   vm::wasm_allocator* alloc;
-   while (stack.pop(alloc)) {
-      delete alloc;
-   }
```

Comment:
I don't think the move to `shared_ptr` is better. This adds overhead simply to remove this destructor. Seems like `unique_ptr` would be a better fit.

Reply:
`unique_ptr` is not allowed. I got 

```
.../libraries/boost/libs/lockfree/include/boost/lockfree/stack.hpp: In instantiation of ‘class boost::lockfree::stack<std::unique_ptr<eosio::vm::wasm_allocator> >’:
.../libraries/chain/include/eosio/chain/wasm_alloc_pool.hpp:49:106:   required from here
.../libraries/boost/libs/lockfree/include/boost/lockfree/stack.hpp:73:58: error: static assertion failed: boost::is_copy_constructible<T>::value
   73 |     BOOST_STATIC_ASSERT(boost::is_copy_constructible<T>::value);
```

Reply:
Then the naked pointer approach is likely better. Not sure we want to pay for the `shared_ptr`. Also the use of `shared_ptr` implies these are shared, but you explicitly do not share them.

Reply:
> I can think of an impl that works for `aquire()` but not one that works with `release()` because `push()` seems to always make a copy?

oh heh, that doesn't matter, since now that I look at the error there is a static assert that says just that

Reply:
Done by https://github.com/AntelopeIO/spring/pull/1378/commits/cc7f17bc0c691d7635c6da566e46fb5caea9b5b7